src/codemagic/models/xctests/converter.py

```diff
@@ -140,7 +174,169 @@ def actions_invocation_record_to_junit(cls, actions_invocation_record: ActionsIn
-        return cls.actions_invocation_record_to_junit(actions_invocation_record)
+    def _iter_nodes(cls, root_node: XcTestNode, node_type: XcTestNodeType) -> Iterator[XcTestNode]:
+        if root_node.node_type is node_type:
+            yield root_node
+        else:
+            for child in root_node.children:
+                yield from cls._iter_nodes(child, node_type)
+
+    @classmethod
+    def _get_run_destination(cls, root_node: XcTestNode) -> Optional[XcDevice]:
+        # TODO: support multiple run destinations
+        #  As a first iteration only one test destination is supported as in legacy mode
+
+        parent: Union[XcTests, XcTestNode] = root_node
+        while isinstance(parent, XcTestNode):
+            parent = parent.parent
+
+        tests = cast(XcTests, parent)
+        if not tests.devices:
+            return None
+        return tests.devices[0]
+
+    @classmethod
+    def _get_test_suite_name(cls, xc_test_suite: XcTestNode) -> str:
+        name = xc_test_suite.name or ""
+        device_info = ""
+
+        device = cls._get_run_destination(xc_test_suite)
+        if device:
+            platform = re.sub("simulator", "", device.platform, flags=re.IGNORECASE).strip()
+            device_info = f"{platform} {device.os_version} {device.model_name}"
+
+        if name and device_info:
+            return f"{name} [{device_info}]"
+        return name or device_info
+
+    @classmethod
+    def _get_test_case_error(cls, xc_test_case: XcTestNode) -> Optional[Error]:
+        if xc_test_case.result is not XcTestResult.FAILED:
+            return None
+
+        failure_messages_nodes = cls._iter_nodes(xc_test_case, XcTestNodeType.FAILURE_MESSAGE)
+        failure_messages = [node.name for node in failure_messages_nodes if node.name]
+        return Error(
+            message=failure_messages[0] if failure_messages else "",
+            type="Error" if any("caught error" in m for m in failure_messages) else "Failure",
+            error_description="\n".join(failure_messages) if len(failure_messages) > 1 else None,
+        )
+
+    @classmethod
+    def _get_test_case_skipped(cls, xc_test_case: XcTestNode) -> Optional[Skipped]:
+        if xc_test_case.result is not XcTestResult.SKIPPED:
+            return None
+
+        failure_messages_nodes = cls._iter_nodes(xc_test_case, XcTestNodeType.FAILURE_MESSAGE)
+        skipped_message_nodes = (node for node in failure_messages_nodes if node.result is XcTestResult.SKIPPED)
+        skipped_messages = [node.name for node in skipped_message_nodes if node.name]
+
+        return Skipped(message="\n".join(skipped_messages))
+
+    @classmethod
+    def _get_test_node_duration(cls, xc_test_case: XcTestNode) -> float:
+        if not xc_test_case.duration:
+            return 0.0
+
+        duration = xc_test_case.duration.replace(",", ".")
+        if duration.endswith("s"):
+            duration = duration[:-1]
+        return float(duration)
```

Comment:
```suggestion
        return float(re.sub(r"[^\d\.,]", "", xc_test_case.duration).replace(",", "."))
```

Reply:
I'd rather not use this loose regex pattern. It can do transformations that are not intentional and hide values which we're currently not supporting. Current "dumb" approach can handle only strings that contain seconds, but I really have no idea whether Xcode represents longer lasting tests (more than couple of minutes) in some other way. When unsupported string representation comes, then this raises an error and hopefully we'll get to know about it in one way or another.